### PR TITLE
Dart 2 site banner announcing Dart 2

### LINKED ISSUE #759: Travis job link refers to wrong domain
For example, as of the time of creating this issue, the job link URL is: https://github.com/dart-lang/site-www/jobs/362924617

It should refer to travis-ci.org, but doesn't because site-www is missing a shared Jekyll plugin `regex_replace_filter.rb`.

The plan is to fix this once we have site repo sharing setup.

### PR DESCRIPTION
**Pull request by chalin**
Add a Dart 2 announcement banner to very page. This PR is the Dart 2 counterpart to the 1.x #764.

> screen shot 2018-04-09 at 12 44 55

Staged (this _shows_ the banner; merging this PR will NOT show the banner until the sites are switched):
- https://dartlang-org-staging-1.firebaseapp.com
- https://dartlang-org-staging-1.firebaseapp.com/dart-2

Fixes #759

cc @kevmoo

### CODE REVIEW

**Comment by kevmoo** [comment]
Nice!

File: src/_includes/banner.html [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
-        <a target="_blank" 
-          href="https://www.google.com/about/careers/jobs#t=sq&q=j&li=20&l=false&j=dart&jcoid=7c8c6665-81cf-4e11-8fc9-ec1d6a69120c&jcoid=e43afd0d-d215-45db-a154-5386c9036525">Become
-	    a Developer Advocate</a>
-    </p>
+{% if site.url == 'https://www.dartlang.org' %}
+{% comment %}Drop the div .alert classes to revert to the default banner styling.{% endcomment -%}
+<div class="banner alert alert-info">
+  <p class="banner__text">
+    Welcome to <b>Dart 2</b>!
+    Looking for Dart 1.x? See the site
```
Comment (kwalrath) [inline src/_includes/banner.html]: How about:

Welcome to **Dart 2**! Still using Dart 1.x? See the [archived site]() or the [migration guide]().

Comment (kwalrath) [reply src/_includes/banner.html]: It feels like Dart 2 should be a link to something. But what?

Comment (chalin) [reply src/_includes/banner.html]: Changes done and staging server refreshed.

**Comment by chalin** [comment]
> It feels like Dart 2 should be a link to something. But what?

Well, how about making `/dart-2` a _Welcome to Dart 2!_ page, with the following sections:

- Welcome to Dart 2!
  - What's new (the content of what is currently named "Differences")
  - Migrating from Dart 1.x
  - More resources

**Comment by kevmoo** [comment]
SGTM

**Comment by chalin** [comment]
@kwalrath - how about if we merged this as is and tackle the (slight) /dart-2 page reorg in a followup PR?

**Comment by chalin** [comment]
Since @kwalrath gave a thumbs up, I've reworked `/dart-2`.
See https://dartlang-org-staging-1.firebaseapp.com/dart-2

**Comment by kwalrath** [comment]
Let's hold off on changing the focus of /dart-2 for now.

**Comment by chalin** [comment]
@kwalrath - the commit with the /dart-2 edits has been removed. It this ready to go?

**Comment by chalin** [comment]
I've added a snapshot to the opening comment. The staging server has been refreshed too.

**Comment by chalin** [comment]
@kwalrath I've made the changes you requested earlier (but I missed):

> archive -> archived
> , or -> or

I've updated the staging server and the snapshot in the opening comment.

**Pull request merged**